pkg/prom/collector.go

```diff
@@ -1323,6 +1326,51 @@ func (c *SolidfireCollector) collectVolumeAccessGroups(ctx context.Context, ch c
 	return nil
 }
 
+func (c *SolidfireCollector) collectVirtualVolumeTasks(ctx context.Context, ch chan<- prometheus.Metric) error {
+	vvt, err := c.client.ListVirtualVolumeTasks(ctx)
+	if err != nil {
+		return err
+	}
+	mu.Lock()
+	defer mu.Unlock()
+	ch <- prometheus.MustNewConstMetric(
+		MetricDescriptions.VirtualVolumeTasks,
+		prometheus.CounterValue,
+		float64(len(vvt.Result.Tasks)),
+	)
+	return nil
+}
+
+func (c *SolidfireCollector) collectBulkVolumeJobs(ctx context.Context, ch chan<- prometheus.Metric) error {
+	btj, err := c.client.ListBulkVolumeJobs(ctx)
+	if err != nil {
+		return err
+	}
+	mu.Lock()
+	defer mu.Unlock()
+	ch <- prometheus.MustNewConstMetric(
+		MetricDescriptions.BulkVolumeJobs,
+		prometheus.CounterValue,
+		float64(len(btj.Result.BulkVolumeJobs)),
+	)
+	return nil
+}
+
+func (c *SolidfireCollector) collectAsyncResults(ctx context.Context, ch chan<- prometheus.Metric) error {
+	ar, err := c.client.ListAsyncResults(ctx)
```

Comment:
Also, we're primarily interesting in currently running tasks but ListAsyncResults includes tasks that have been completed (Solidfire retains those for some time or up to some limited number).   However, it could also be helpful to label those with `success: false` so maybe we don't want to filter out all the completed ones but just add more labels so that we can build dashboards to show the active ones and the failed ones?

Reply:
Done, also tracking `completed`